lib/checkPullRequestLabels.js

```diff
@@ -0,0 +1,40 @@
+module.exports.checkPullRequestLabels = async function(context) {
+  var pullRequest = context.payload.pull_request;
+  var pullRequestNumber = pullRequest.number;
+
+  // eslint-disable-next-line no-console
+  console.log('RUNNING LABEL CHECKS ON PULL REQUEST ' + pullRequestNumber + ' ...');
+
+  var labels = pullRequest.labels;
+  var userName = pullRequest.user.login;
+  var projectOwnerLabel;
+  var hasProjectOwnerLabel = labels.some(function(label) {
+    projectOwnerLabel = label.name;
+    return projectOwnerLabel.startsWith('PROJECT');
+  });
+
+  // If the PR has a project owner label, assign the project owner
+  // for the first-pass review of the PR and leave a top-level comment.
+  if(hasProjectOwnerLabel === true) {
+    var labelSubstrings = projectOwnerLabel.split('@');
+    var projectOwner = labelSubstrings[labelSubstrings.length-1].trim();
+
+    var assigneeParams = context.issue({assignees: [projectOwner]});
+    await context.github.issues.addAssigneesToIssue(assigneeParams);
+
+    var commentParams = context.issue({body: 'Assigning @' + projectOwner +
+      ' for the first-pass review of this pull request. Thanks!'});
+    await context.github.issues.createComment(commentParams);
+    return;
+  }
+
+  // If no project owner label is found, close the PR.
+  // Ask the author to add the label and re-open it.
+  var stateParams = context.issue({state: 'closed'});
+  await context.github.issues.edit(stateParams);
+
+  var stateCommentParams = context.issue({body: 'Hi, @' + userName +
+    '. This pull request has been closed since it does have a project ' +
```

Comment:
Also, not entirely sure whether closing is too drastic. But I think that's a question for you + Sandeep + dev workflow team; I'm happy to defer to the joint decision. Maybe set up a quick group chat and ask.

Reply:
I have already discussed with Sandeep and we've decided to close the pull request so that all open PRs always follow the review cycle and the process remains streamlined.

Reply:
OK, sounds good, happy to go with what you think best. Maybe document the design decisions in the PR description.

Reply:
I have modified the corresponding issue description. I'll go ahead and add it here as well. Thanks!